src/attributes.jl

```diff
@@ -388,7 +388,7 @@ error is thrown if it supports the submittable `sub` but it cannot be submitted.
 function submit end
 function submit(model::ModelLike, sub::AbstractSubmittable, args...)
     if supports(model, sub)
-        throw(SubmitNotAllowed(sub))
+        throw(MethodError(submit, (model, sub, args...)))
```

Comment:
This change doesn't address the concern that a JuMP user will get an MOI MethodError if they pass the wrong number of arguments (and there's no obvious way for JuMP to fix this).